.devcontainer/container_post_create.sh

```diff
@@ -12,3 +12,6 @@ if [ -n "\$DISPLAY" ]; then
 fi
 EOF
 fi
+
+USER=batman
+sudo chown -R $USER: /tmp/scons_cache
```

Comment:
This is not necessary if you create the directories in their current location in the Dockerfile after switching users:
```
USER $USER
...
RUN mkdir /tmp/comma_download_cache
RUN mkdir /tmp/scons_cache
```
This is better because it has the benefit of things working if you use the docker container outside of using vscode dev containers.

Reply:
These are actually volume mounted in, so what happens to me here was my old dev container was running as root, it created files as root, and then I switched to batman and the volume still had files as root. Perhaps we should leave this so we don't break the devcontainer for existing users?

Reply:
I think that is just a question of when we should remove these lines from `.devcontainer/container_post_create.sh`.  I assume backward compatibility is only necessary for a user until this script runs once.  If we want to support backward compatibility I don't think we want to support it forever, and moving the directory creation to the `Dockerfile` is a prerequisite, so I think it should still be done now.  If we keep these lines in this file I would also add a comment explaining this is for backwards compatibility and that we should delete this in the future.

Reply:
Added a comment about how it should be temporary. But adding it to the dockerfile won't do anything AFAIK because it's volume mounted in (to preserve the cache across devcontainer restarts), so it will just copy the chown of the host

Reply:
OK yeah I see now it is fine.  I thought they were declared in the Dockerfile using the `VOLUME` keyword (which always creates as root) and it wouldn't work if you run the container as a non-root user and didn't mount in anything, but I see  we don't declare these directories `VOLUME` in the Dockerfile and because /tmp is world writable everything does work.